src/App.js

```diff
@@ -1,24 +1,39 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from "react";
+import Header from "./component/header/header/Header";
+import Main from "./component/main/Main";
+import { BrowserRouter, Route, Routes } from "react-router-dom";
+import Contact from "./component/Contact";
+import About from "./component/About";
 
 function App() {
+  const [allArticles, setAllArticles] = useState([]);
+  const [searchByToic, setSearchTopic] = useState("");
```

Comment:
### state is very high. Does it need to be this high?

Does the NavBar really need the state for example?

I'd recommend lowering state, keep the articles in the articles component for example